### PR TITLE
Add support for transformation max depths

### PR DESCRIPTION
**Pull request by rubenvanassche**

### CODE REVIEW

**Review by sebastiandedeyne — CHANGES_REQUESTED** [review]

File: config/data.php [hunk 1]
```diff
@@ -120,6 +120,21 @@
...
+     * package will not enforce a maximum depth.
+     */
+    'max_transformation_depth' => null,
+
+    /**
+     * When the maximum transformation depth is reached, the package will throw an exception.
+     * You can disable this behaviour by setting this option to true which will return an
+     * empty array.
+     */
+    'fail_when_max_transformation_depth_reached' => true,
```
[the remark this answers is not in the document]
Comment (rubenvanassche) [reply config/data.php:136]: Jup

File: src/Resolvers/TransformedDataResolver.php [hunk 2]
```diff
@@ -20,6 +21,8 @@
 
 class TransformedDataResolver
 {
+    use ChecksTransformationDepths;
```
[the remark this answers is not in the document]
Comment (rubenvanassche) [reply src/Resolvers/TransformedDataResolver.php:24]: It needs to be somewhere, also makes it easy for other people who want to build casts based upon max depth checks to include it.

**Comment by inmanturbo** [comment]
@rubenvanassche 
It looks like this broke some packages that depend on laravel-data.
```
Spatie\LaravelData\Support\Transformation\TransformationContext::__construct(): Argument #11 ($throwWhenMaxDepthReached) must be of type bool, null given
```
It looks like updating the config to include ` 'throw_when_max_transformation_depth_reached' => true,` will probably fix it at the app level, but I'm not sure how to fix the packages' ci.

Any advice?

**Comment by inmanturbo** [comment]
for now I'm adding 

```php
config()->set('data.throw_when_max_transformation_depth_reached', true);
```
to `getEnvironmentSetUp()` in the test case and it seems to work but it feels strange to have to explicitly set the config for a dependency to its default value. These packages don't have any published laravel-data configuration nor do they interact with its config.

**Comment by rubenvanassche** [comment]
See the answer here: https://github.com/spatie/laravel-data/issues/731

**Pull request merged**